### PR TITLE
Add medical canisters

### PR DESCRIPTION
**Pull request by chuck-stein**
Also updated exit portal prefab because previous one worked better as a healing particle effect and new one looks better

### CODE REVIEW

File: Assets/Scripts/AsteroidCollisionHealthEffect.cs [hunk 1]
```diff
@@ -1,5 +1,4 @@
-﻿using System;
-using UnityEngine;
+﻿using UnityEngine;
 
 public class AsteroidCollisionHealthEffect : HealthEffectBehavior
 {
```
Comment (Unix-Code) [inline Assets/Scripts/AsteroidCollisionHealthEffect.cs:4]: I know this wasn't touched by this PR, but if you could quickly fix the typo, that'd be great:
On line 11 in your version, size is set to `size.magnitude` when it really should be `size.x`. The calculation is a bit "off" and skews towards higher than expected.

File: Assets/Scripts/AsteroidMovement.cs [hunk 2]
```diff
@@ -16,14 +16,8 @@ private void Start()
         speed = Random.Range(minSpeed, maxSpeed);
         appliedInitialForce = false;
         rb = GetComponent<Rigidbody>();
```
Comment (Unix-Code) [inline Assets/Scripts/AsteroidMovement.cs]: Can you revert the changes in this file? I already fix this in my ProceduralGeneration branch, so changing it here would just lead to some annoying merge conflicts.

File: Assets/Scripts/BlackHolePull.cs [hunk 3]
```diff
@@ -30,6 +30,9 @@ private void OnTriggerExit(Collider other)
 
     private void FixedUpdate()
     {
+        // first filter out any pulled bodies that may have been destroyed since the last update
```
Comment (Unix-Code) [inline Assets/Scripts/BlackHolePull.cs:33]: We'd need to performance test this. But thanks for adding this in.

File: Assets/Scripts/MedicalCanisterPickup.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+public class MedicalCanisterPickup : MonoBehaviour
+{
+
+    public GameObject healAura;
+
+    private void OnTriggerEnter(Collider other)
+    {
+        if (!other.CompareTag("Player")) return;
+        other.GetComponentInChildren<GrappleGunBehavior>().StopGrapple();
```
Comment (Unix-Code) [inline Assets/Scripts/MedicalCanisterPickup.cs]: Instead of dynamically getting this component, it might be worth setting up this reference on start.

File: Assets/Scripts/RandomStartAngle.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+﻿using UnityEngine;
+
+public class RandomStartAngle : MonoBehaviour
+{
+
+    private void Start()
+    {
```
Comment (Unix-Code) [inline Assets/Scripts/RandomStartAngle.cs]: `Random.rotation` returns a random quaternion. No need to do this.

File: Assets/Scripts/Rotation.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿using UnityEngine;
```
Comment (Unix-Code) [inline Assets/Scripts/Rotation.cs]: Can you do this via an idle animation instead of a monobehavior?

File: Assets/Scripts/SelfDestruct.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿using UnityEngine;
```
Comment (Unix-Code) [inline Assets/Scripts/SelfDestruct.cs:1]: How is this used? Are health pickups generated at world generation or do they constantly spawn...?

**Review by Unix-Code — CHANGES_REQUESTED** [review]
Open questions and change requests

File: Assets/Scripts/AsteroidCollisionHealthEffect.cs (hunk 1 above)
Comment (chuck-stein) [reply Assets/Scripts/AsteroidCollisionHealthEffect.cs:4]: Okay I can change it for now but honestly that calculation should be completely different anyway. The damage dealt on collision should be proportional to the speed of impact, not the size of the asteroid.

File: Assets/Scripts/BlackHolePull.cs (hunk 3 above)
Comment (chuck-stein) [reply Assets/Scripts/BlackHolePull.cs:33]: Is it good for now though? I didn't notice any performance issues.

File: Assets/Scripts/SelfDestruct.cs (hunk 7 above)
Comment (chuck-stein) [reply Assets/Scripts/SelfDestruct.cs:1]: It's placed on the particle effect instantiated when you pickup a medical canister so that the gameobject is destroyed once the particle effect ends

File: Assets/Scripts/AsteroidMovement.cs (hunk 2 above)
Comment (chuck-stein) [reply Assets/Scripts/AsteroidMovement.cs]: Done

File: Assets/Scripts/MedicalCanisterPickup.cs (hunk 4 above)
Comment (chuck-stein) [reply Assets/Scripts/MedicalCanisterPickup.cs]: Done

File: Assets/Scripts/RandomStartAngle.cs (hunk 5 above)
Comment (chuck-stein) [reply Assets/Scripts/RandomStartAngle.cs]: Removed this file since you'll take care of it in procedural generation

File: Assets/Scripts/Rotation.cs (hunk 6 above)
Comment (chuck-stein) [reply Assets/Scripts/Rotation.cs]: Done

File: Assets/Scripts/BlackHolePull.cs (hunk 3 above)
Comment (Unix-Code) [reply Assets/Scripts/BlackHolePull.cs:33]: Yeah it's probably fine but maybe not great when there are a ton of asteroids being pulled all the time, and this is running every fixed update

File: Assets/Scripts/GrappleGunBehavior.cs [hunk 8]
```diff
@@ -109,7 +124,7 @@ public bool IsGrappling()
     // TODO: make this private if nothing outside the class uses it
     public Vector3 GrapplePoint()
     {
-        if (grappleObjOffset == null) throw new InvalidOperationException("There is no active grapple point.");
-        return grappleObj.position + grappleObjOffset.Value;
+        if (grappledObjOffset == null) throw new InvalidOperationException("There is no active grapple point.");
```
Comment (Unix-Code) [inline Assets/Scripts/GrappleGunBehavior.cs]: Use HasValue instead of == null

**Review by Unix-Code — APPROVED** [review]

File: Assets/Scripts/GrappleGunBehavior.cs (hunk 8 above)
Comment (chuck-stein) [reply Assets/Scripts/GrappleGunBehavior.cs]: Done
